cogs/notification.py

```diff
@@ -89,25 +91,39 @@ async def notifier(self, itn : discord.Interaction, username: str, channel: disc
             The username of the twitter user you want to turn off notifications for.
         channel: discord.TextChannel
             The channel which set to delivers notifications.
-        """
-        
+        """  
         await itn.response.defer(ephemeral=True)
         
         conn = sqlite3.connect(f"{os.getenv('DATA_PATH')}tracked_accounts.db")
-        conn.row_factory = sqlite3.Row
         cursor = conn.cursor()
         
-        cursor.execute('SELECT user_id FROM notification, user WHERE username = ? AND channel_id = ? AND user_id = id', (username, str(channel.id)))
-        match_notifier = cursor.fetchone()
-        if match_notifier != None:
-            cursor.execute('UPDATE notification SET enabled = 0 WHERE user_id = ? AND channel_id = ?', (match_notifier['user_id'], str(channel.id)))
-            conn.commit()
-            await itn.followup.send(f'successfully remove notifier of {username}!', ephemeral=True)
-        else:
-            await itn.followup.send(f'can\'t find notifier {username} in {channel.mention}!', ephemeral=True)
+        server_id = str(itn.guild_id)
+
+        cursor.execute(f"SELECT * FROM user WHERE username='{username}'")
+        match_user = cursor.fetchone()
+        
+        if match_user is None:
+            await itn.followup.send(f'can\'t find notifier{username}', ephemeral=True)
+            return
         
+        await self.account_tracker.removeTask(username)
+        
+        cursor.execute(f"DELETE FROM user WHERE username='{username}'")
```

Comment:
bug: deleting data from DB is dangerous, this feature caused when the same user was deleted on other servers, although there were other servers that turned on notifications for the user, the user was deleted. The notification should be disabled(set enabled to 0) rather than deleted. Make sure to deactivate the user when all servers have disabled notifications for the user. Perhaps we should add an enabled attribute to the user?
![image](https://github.com/Yuuzi261/Tweetcord/assets/63157008/ba809722-d671-4879-8875-b5aaa3de57b8)
![image](https://github.com/Yuuzi261/Tweetcord/assets/63157008/fda3007b-5aa3-4daf-bbf3-7da3fabd5368)